### PR TITLE
Added workflows

### LINKED ISSUE #14: Add GitHub actions CI/CD for linting and testing
Add Actions to the repository in order to automate testing and linting for pull requests

### PR DESCRIPTION
**Pull request by Alxjlv**
Closes #14 

Both frontend and backend CI yml files added to run tests and linting

### CODE REVIEW

File: .github/workflows/backend.yml [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+      - name: Install Node.js dependencies
+        run: npm ci
+
+      - name: Run linter
+        run: npm run lint
+
+      - name: Run tests
+        run: |
+            npm run build --if-present
+            npm test
```
Comment (taitfuller) [inline .github/workflows/backend.yml:36]:
```suggestion
            npm test
```
```suggestion
            npm test
            
```

**Review by taitfuller — CHANGES_REQUESTED** [review]
Overall looks good! Prettier is setup for the repository root as of #18 but if you could run `prettier --write .` on these that'd be great!

**Comment by Alxjlv** [comment]
> Overall looks good! Prettier is setup for the repository root as of #18 but if you could run `prettier --write .` on these that'd be great!

@taitfuller I'm not entirely sure what you mean by running `prettier --write .` - wouldn't that go against the purpose of the action? None of the write changes would be saved to the repo, and if changes are required, then that should be a fail state.

Is there a prettier command I can use to see if `--write` was somehow missed?

**Comment by taitfuller** [comment]
Ah sorry! I meant to run prettier on the code that you're submitting so that the `.yml` files are formatted consistently by Prettier. All future commits after #18 will have this occur automatically with the precommit hook 👌🏼

**Review by taitfuller — APPROVED** [review]
Looks great!

**Pull request merged**